### DISCUSSION TITLE
2020/04/29/Print-call-stack-in-AOSP-native-code

### DISCUSSION THREAD

**Discussion by giscus** [utzcoz/utzcoz.github.io#31] [Comments]
# 2020/04/29/Print-call-stack-in-AOSP-native-code

This article based on AOSP 9.0.

https://utzcoz.github.io/2020/04/29/Print-call-stack-in-AOSP-native-code.html

**Comment by sysescool** [comment]
BTW, I try it in AOSP 11.
You should Add `libutils_headers` to your `header_lib` of `Android.bp` too.

**Reply by sysescool** [reply]
callstack.h location:
`aosp_11/system/core/libutils/include/utils/CallStack.h`
Android.bp location:
`aosp_11/system/core/libutils/Android.bp`

what is written in Android.bp:
```
cc_library_headers {
    name: "libutils_headers",
    //............
}
cc_library {
    name: "libutilscallstack",
    //............
        srcs: [
        "CallStack.cpp",
    ],
    //....................
}
```

**Reply by utzcoz** [reply]
@sysescool Thanks. Could you help to send a PR to add these extra information for Android 11?

**Reply by sysescool** [reply]
@utzcoz my pleasure, but how to send a PR? I mean this is a article of you, I am not sure to send a PR from where to where.

**Reply by utzcoz** [reply]
@sysescool You can send a PR to modify https://github.com/utzcoz/utzcoz.github.io/blob/master/_posts/2020-04-29-Print-call-stack-in-AOSP-native-code.markdown. It is the source of this article. You can read https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-pull-requests for PR in GitHub. If this has any difficulty for you, please let me know. I also can be the backup to update this article this weekend.

**Reply by utzcoz** [reply]
@sysescool I updated this article with extra steps for AOSP 11. Thanks for your contribution.
